parachain/pallets/basic-channel/rpc/src/lib.rs

```diff
@@ -0,0 +1,44 @@
+use jsonrpc_core::{Error, ErrorCode, Result};
+use jsonrpc_derive::rpc;
+
+use codec::Encode;
+use sp_core::H256;
+use sp_runtime::{offchain::storage::StorageValueRef, traits::Keccak256};
+
+use snowbridge_basic_channel_merkle_proof::merkle_proof;
+use snowbridge_basic_channel_primitives::StoredLeaves;
+
+pub struct BasicChannel;
+impl BasicChannel {
+	pub fn new() -> Self {
+		Self {}
+	}
+}
+
+#[rpc]
+pub trait BasicChannelApi {
+	#[rpc(name = "basicChannel_outbound_getMerkleProof")]
+	fn get_merkle_proof(&self, commitment_hash: H256, leaf_index: u64) -> Result<Vec<u8>>;
+}
+
+impl BasicChannelApi for BasicChannel {
+	fn get_merkle_proof(&self, commitment_hash: H256, leaf_index: u64) -> Result<Vec<u8>> {
+		let oci_mem = StorageValueRef::persistent(&commitment_hash.as_bytes());
+
+		if let Ok(Some(StoredLeaves(leaves))) = oci_mem.get::<StoredLeaves>() {
```

Comment:
I guess you could also replace this with a `match`?

Reply:
We could do, but we're only matching the one pattern, which is the use-case for `if let ... else`. If we had a second pattern (or more) we were interested in, we'd definitely prefer `match`.